masonry/src/contexts.rs

```diff
@@ -443,6 +443,7 @@ impl_context_method!(MutateCtx<'_>, EventCtx<'_>, LifeCycleCtx<'_>, {
     /// Request an animation frame.
     pub fn request_anim_frame(&mut self) {
         trace!("request_anim_frame");
+        self.widget_state.needs_anim = true;
         self.widget_state.request_anim = true;
```

Comment:
I guess we want to either rename the `needs_...` variables to be more clear, or document the difference between `request_...` and `needs_...` well.

At a first glance, the naming suggests a little bit redundancy here, at least it wasn't clear to me what the difference exactly should be.
(But I don't think that this should be done in this PR, and it's probably better to be consistent with the rest of the naming).

Reply:
> I guess we want to either rename the `needs_...` variables to be more clear, or document the difference between `request_...` and `needs_...` well.

Right now the fields all have doc comments that follow this pattern:

- **request_xxx:** "This widget explicitly requested XXX"
- **needs_xxx:** "This widget or a descendant explicitly requested XXX"

Once we do a documentation pass I'll explicitly note the pattern as well, at least in WidgetState and `ARCHITECTURE.md`. If you think there's other places we could cover, I'm interested. This is the sort of pattern where you want redundancy in documenting it.

Although note that it's strictly an internal naming convention that library users won't be exposed to, so it limits the places we're likely to document it in.

I do like those names. I think they're concise, which is valuable, and the "request/needs" difference feels to me like it's easy to remember once it's been explained.